### DISCUSSION TITLE
Support for lights attached to vehicles

### DISCUSSION THREAD

**Discussion by Davidsastresas** [microsoft/AirSim#4329] [Ideas]
I think it would be useful to be able to change the lights attached to vehicles, in order to better estimate simulations when computer vision is involved in a system where it can control lights.

I think I have sorted out how to include permanent lights attached to the player in Unreal, but I don't know how to start attaching that to a connection in Airsim, and thus to the autopilot or software managing the vehicle.

I would not mind to work on this and do a PR, but I would appreciate a couple of hints about where to look in Airsim to make the connection to such element in unreal.

Thanks!

**Comment by rajat2004** [comment]
Recently a feature to control lights was added, see https://github.com/microsoft/AirSim/pull/3991, might be useful here
For a quick hack, might be possible to continuously change the pose of the light object as needed to follow the vehicle, using say `simSetObjectPose`. Better way wold probably be to attach the blueprint to the vehicle itself, but not sure if it would still work through the API

**Comment by Davidsastresas** [comment]
Thank you very much, that is enough for me to keep pulling that string. If I manage to do something nice I will submit a PR about it. Thanks!

**Comment by zimmy87** [comment]
Hi @Davidsastresas, you should be able to achieve what you want by adding one of the light components to the desired vehicle blueprint, then wrapping any operations on that light component with a blueprint function that you then call from a newly added API, similar to what @saihv did with the `simSetLightIntensity` API in #3991

**Reply by Davidsastresas** [reply]
That is amazing, thank you very much for the update. I should be testing this in a custom vehicle very soon. If over the way I do something PR worth I will do my best to prepare it and push it. Thanks!
